crypto/dsa/dsa_pmeth.c

```diff
@@ -77,13 +77,8 @@ static int pkey_dsa_sign(EVP_PKEY_CTX *ctx, unsigned char *sig,
     DSA_PKEY_CTX *dctx = ctx->data;
     DSA *dsa = ctx->pkey->pkey.dsa;
 
-    if (dctx->md) {
-        if (tbslen != (size_t)EVP_MD_size(dctx->md))
-            return 0;
-    } else {
-        if (tbslen != SHA_DIGEST_LENGTH)
-            return 0;
-    }
+    if (dctx->md && tbslen != (size_t)EVP_MD_size(dctx->md))
```

Comment:
Currently it's considered more fashionable to check to NULL explicitly, i.e. as `dctx->md != NULL`.

Reply:
Oops, missed this one. I'll change this.